src/com/seafile/seadroid2/ui/fragment/StarredFragment.java

```diff
@@ -249,5 +261,34 @@ protected void onPostExecute(List<SeafStarredFile> starredFiles) {
         }
     }
 
+    class StarFileTask extends AsyncTask<Void, Void, String> {
+        private String repoId;
+        private String path;
+        private SeafException err;
+
+        public StarFileTask(String repoId, String path) {
+            this.repoId = repoId;
+            this.path = path;
+        }
+
+        @Override
+        protected String doInBackground(Void... params) {
 
+            try {
+                return mActivity.getDataManager().star(repoId, path);
+            } catch (SeafException e) {
+                err = e;
+                return null;
+            }
+        }
+
+        @Override
+        protected void onPostExecute(String s) {
+            if (s == null) {
+                if (err != null)
+                    ToastUtils.show(mActivity, err.getCode() == 201 ? R.string.star_file_succeed : R.string.star_file_failed);
```

Comment:
After change the code in SeafConnection we need to change here too (Don't treat http 201 as exception).